### PR TITLE
Upstream sync 9/N: merge dc5101fb1b replace batch_norm to numerically identical without cudnn (conflict)

### PR DESCRIPTION
**Pull request by roberteg16**
## Context

Ninth step of the batched upstream catch-up. Stacked on #<batch 8 PR>.

**Conflict-only** step.

Merged upstream changes:
1. `dc5101fb1b` #51734 — replace batch_norm to numerically identical without cudnn

One conflict, in `vllm/model_executor/models/vision.py`, and it is the most dangerous shape in this series: an upstream rewrite whose deletion hunk overlaps a fork *append*, where taking "ours" compiles, lints, and returns silently wrong numbers.

Upstream replaces `F.batch_norm` in `PixelsNormalizer.forward` with an equivalent broadcasted multiply-add (cuDNN caps the batch dim near 65535; patches exceed it on large image-heavy requests → `CUDNN_STATUS_INTERNAL_ERROR`). In doing so it renames the local: the fork rebound `grid_thw` in place, upstream introduces `x`.

```
ours:     return grid_thw.view(patches, size).to(visual_dtype)
          <+30 lines: use_npu_vision_backend() / get_npu_vision_backend()>
theirs:   return x.view(patches, size).to(visual_dtype)
resolved: return x.view(patches, size).to(visual_dtype)          <- theirs
          <+30 lines: use_npu_vision_backend() / ...>            <- ours
```

Both halves, in that order. Note this is *not* a simple union of the hunk as git presented it: the fork's return line is discarded while the fork's appended block is kept.

git could not align these because the fork had appended an NPU-backend section immediately after the return, so the two sides' return statements landed in different hunks with the fork's block between them.

Resolved by taking upstream's return **and** keeping the fork's NPU block.

## Audit

The union is not optional here; both halves are load-bearing, in opposite directions.

- **Keeping the fork's return line is the trap.** The merged body computes the normalization into `x` and leaves `grid_thw` holding the *raw input*, so `return grid_thw.view(...)` returns un-normalized pixels. It is valid Python, it is `ruff`-clean, and it silently produces wrong vision embeddings on every multimodal request — no error anywhere. This is precisely the "clean auto-merge is not proof of a correct merge" case.
- **Dropping the fork's NPU block would be an `ImportError` at load.** Both functions have live callers: `use_npu_vision_backend` in `vllm/model_executor/models/qwen2_5_vl.py:1152` and `get_npu_vision_backend` in `vllm/model_executor/models/qwen2_5_vl_npu.py:20`.

Checked for leftovers from the rewrite, since upstream deleted the only `F.batch_norm` call: no `F.` usages and no `import torch.nn.functional as F` remain in the file (only the explanatory comment mentions it), and the `running_mean`/`running_var` buffers the old call needed are gone too — upstream removed them in the same commit and the merge kept that removal. Had either been left behind, `ruff` would have flagged the import but *not* the dead buffers.

`py_compile`, `ruff check` and `ruff format` pass on the merged file.

**Merge commit only — do not squash or rebase.**

AI assistance was used to prepare this merge.

## Test plan

- [x] Return statement traced to the normalized local; un-normalized-input trap identified and avoided
- [x] Both NPU functions confirmed to have live callers before being retained
- [x] `F` import and `running_mean`/`running_var` buffers confirmed fully removed
- [x] `py_compile` + `ruff check` + `ruff format` on the merged file
- [x] Multimodal smoke test on gfx1151 (a vision model through `PixelsNormalizer`) — run 3512149: 4 VLMs (gemma-3-4b, gemma-4-31B, Qwen3-Omni-30B, Qwen3.6-35B) PASS with sanity checks green
- [x] Build + 6-model benchmark sweep vs the Strix Halo dashboard — run 3512149, no regression (see comment)

### CODE REVIEW

**Comment by roberteg16** [comment]
## Benchmarks

Six-model sweep run on the Strix Halo board via [Custom Regression run 3512149](https://gitenterprise.xilinx.com/FaaSApps/rocm-scripts/actions/runs/3512149).

**No regression** — decode and TTFT both flat across the set.

| bench | model | decode tok/s | baseline | Δ | TTFT ms | baseline | Δ |
|---|---|---:|---:|---:|---:|---:|---:|
| bench1 | Qwen3-30B-A3B-Instruct-2507-AWQ-4bit | 80.2 | 80.2 | +0.1% | 187.3 | 187.0 | +0.2% |
| bench2 | Qwen3.6-35B-A3B-W4A16 (int8:g32 lm-head) | 82.9 | 83.0 | −0.1% | 265.8 | 281.5 | −5.6% |
| bench3 | gemma-3-4b-it-quantized.w4a16 | 61.6 | 61.5 | +0.2% | 464.2 | 463.8 | +0.1% |
| bench4 | Qwen3-Omni-30B-A3B-Instruct-AWQ-4bit | 75.5 | 75.9 | −0.6% | 688.3 | 702.1 | −2.0% |
| bench5 | Qwen2.5-0.5B-Instruct-AWQ | 357.0 | 358.5 | −0.4% | 16.9 | 16.8 | +0.2% |
| bench6 | gemma-4-31B-it-AWQ-4bit | 10.3 | 10.3 | +0.1% | 1536.0 | 1533.7 | +0.2% |

Largest decode move is −0.6%, inside the 1% gate. All six report `PASS` with sanity checks green and exit code 0.

Decode is derived from `median_tpot_ms` rather than the report's `max_output_tokens_per_s`, which is rounded to whole tok/s — at bench5's ~357 tok/s one integer step is already 0.3%, so that field cannot resolve a 1% gate.

Baselines are the nightly dashboard run of 2026-09-05 (`20260905_180640`, vLLM `c84454ff4`).

**Scope.** The run built `4f03b854e`, GitHub's merge commit for #1259 (`9686e281df` into `0a7aeafd4d`) — the top of batch 12, so it contains this PR and every batch below it, and **excludes batch 13 (#1260)**. Individual batches were not swept separately; a regression introduced by one and offset by another would not be visible here, though nothing in the numbers suggests that.

Environment: host `xconucstrhalo07`, gfx1151, 96 GiB. ROCm `10.1.0a20260906`, torch 2.13.0, triton `3.8.0+git4cff872c`, transformers 5.14.1.

AI assistance was used to run this verification.

**Review by mgehre-amd — APPROVED** [review]
